versions/1.21.5/src/main/java/at/hannibal2/skyhanni/utils/compat/OrdereredTextUtils.kt

```diff
@@ -57,29 +56,52 @@ object OrderedTextUtils {
     fun legacyStringToStringVisitable(legacyString: String): StringVisitable {
         val segments = mutableListOf<StringVisitable>()
         var lastStyle = Style.EMPTY
-        var wasLastStyle = false
+
         val sb = StringBuilder()
 
-        for (char in legacyString) {
+        var index = 0
+        while (index < legacyString.length) {
+            val char = legacyString[index]
+
             if (char == '§') {
-                wasLastStyle = true
-            } else if (wasLastStyle) {
-                if (sb.isNotEmpty()) {
+
+                if (index + 1 < legacyString.length) {
+
                     segments.add(StringVisitable.styled(sb.toString(), lastStyle))
+
                     sb.clear()
-                }
 
-                val formatting = Formatting.byCode(char)
-                if (formatting != null) {
-                    lastStyle = lastStyle.withExclusiveFormatting(formatting)
-                } else if (char == 'z') {
-                    lastStyle = lastStyle.withColor(CHROMA_COLOR)
+                    val nextChar = legacyString[index + 1]
+
+                    if (legacyString[index + 1] == '#') {
+
+                        if (index + 8 < legacyString.length) {
+                            val hexColor = legacyString.substring(index + 2, index + 8)
+
+                            hexColor.toIntOrNull(16) ?: { color: Int ->
```

Comment:
This one also needs to be updated
```suggestion
                            hexColor.toIntOrNull(16) ?.let { color: Int ->
```
(I thought `.also` was the right one to use here instead of `.let` because nothing's being returned, but maybe not. dosent really matter anyways.)